misc/codegen/generators/qlgen.py

```diff
@@ -316,39 +316,27 @@ def _get_stub(cls: schema.Class, base_import: str, generated_import_prefix: str)
                    ql_internal="ql_internal" in cls.pragmas)
 
 
-def _patch_class_qldocs(cls: str, qldoc: str, stub_file: pathlib.Path):
+_stub_qldoc_header = "// the following QLdoc is generated: if you need to edit it, do it in the schema file\n"
+
+_class_qldoc_re = re.compile(rf"(?P<qldoc>(?:{_stub_qldoc_header})?/\*\*.*?\*/\s*|^\s*)class\s+(?P<class>\w+)",
+                             re.MULTILINE | re.DOTALL)
```

Comment:
It might be worth using `re.escape` on `_stub_qldoc_header` in case that string ever contains a regex character.

Otherwise LGTM.